modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java

```diff
@@ -182,18 +192,32 @@ protected void doServeResource(
 		}
 	}
 
-	private JSONArray _getAssetCategoriesJSONArray(
-		ContentDashboardItem contentDashboardItem, Locale locale) {
+	private HashMap<String, Object> _buildVocabularyDetails(
+			Locale locale, AssetCategory assetCategory,
+			AssetVocabulary currentVocabulary)
+		throws PortalException {
 
-		List<AssetCategory> assetCategories =
-			contentDashboardItem.getAssetCategories();
+		return HashMapBuilder.<String, Object>put(
+			"categories",
+			() -> {
+				ArrayList<String> categoriesArray = new ArrayList<>();
 
-		Stream<AssetCategory> stream = assetCategories.stream();
+				categoriesArray.add(assetCategory.getTitle(locale));
```

Comment:
Why do we need to use an array, if we only add one value?

Reply:
We are calling `_build Vocabulary Details` once for each category if it's vocabulary hasn't been seen before.
We add the rest of the categories [here](https://github.com/liferay-tango/liferay-portal/blob/e5f83e7872e3132714349532687d0b9c59b0290a/modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java#L274)